### PR TITLE
Add mongo

### PR DESCRIPTION
**Pull request by DevDelvo**
Added MongoDB connection and a basic user name with email and password model.  Also added dotenv for .env variables. Had some issues with babel and had to upgrade them to get `npm run dev` to work properly on my machine. Hope that is not an issue.

### CODE REVIEW

File: server/app.js [hunk 1]
```diff
@@ -3,11 +3,24 @@ import express, { json, urlencoded } from "express";
 import { join } from "path";
 import cookieParser from "cookie-parser";
 import logger from "morgan";
+const mongoose = require("mongoose");
+require('dotenv').config();
```
Comment (skassam21) [inline server/app.js]: this is unnecessary since it's in the bin/www file

File: server/app.js [hunk 2]
```diff
@@ -3,11 +3,24 @@ import express, { json, urlencoded } from "express";
...
+
+const { MONGO_URI } = process.env;
```
Comment (skassam21) [inline server/app.js:8]: we should probably have a config file

**Review by skassam21 — COMMENTED** [review]
Looks good - you should get others to review though too - the idea is to have the team review and i'll take a bulk review on all pull requests before standups and make comments

File: server/app.js [hunk 3]
```diff
@@ -19,12 +31,12 @@ app.use("/", indexRouter);
...
 
 // catch 404 and forward to error handler
-app.use(function(req, res, next) {
+app.use(function (req, res, next) {
   next(createError(404));
 });
 
 // error handler
-app.use(function(err, req, res, next) {
+app.use(function (err, req, res, next) {
```
Comment (evankp) [inline server/app.js:39]: More of style choice than a need to add, but we could use ES6 arrow functions here. app.use((req, res, next) => {...})

**Review by evankp — APPROVED** [review]
I'm guessing server/.env was removed to put .env on the .gitignore list

File: server/app.js (hunk 3 above)
Comment (DevDelvo) [reply server/app.js:39]: I agree, I think this was in the boilerplate and my linter just added the spaces.

Comment (evankp) [reply server/app.js:39]: Figured 👍

**Pull request merged**